### DISCUSSION TITLE
How to avoid several standards? publiccode.yml versus Citation.cff

### DISCUSSION THREAD

**Discussion by broeder-j** [publiccodeyml/publiccode.yml#162] [Q&A] [answered]
First, let me thank you for your great work and efforts, and please do not get this as rude (I am far a way from all these efforts).
I was just wondering if and how one can avoid multiple standards in this area:

As a data steward I see currently 2 standards emerging in this area (your are probably aware of this?). publiccode.yml and [Citation.cff.](https://citation-file-format.github.io/)
While Citation.cff is meant for Software and Datasets, publiccode.yml is meant for Software only. Did you get in contact with the Citation.cff team to avoid that there will be two standards in the future? However this is done, maybe at this 'early stage' one could still merge these two? Actually for certain software projects there are already other standards.
For example for python software they finally improve the metadata for the pyproject.toml. So for every python project people I work with double, triple, ... their metadata already (which I think is not nice).

At least in my area Citation.cff is widely used, promoted and supported already by invenio (look at zenodo), github and others.
I very much like what you did and we definitely need rich metadata for software, Citation.cff is currently very minimal here, but it is the broader standard. In principle Citation.cff could just adapt all your allowed metadata if the type is software.

Or do you think there is are good reason to have two or that these two things will develop in totally different directions, i.e that Citation.cff stays minimal and that for software one does a rich publiccode.yml from which one could extract a Citation.cff?

**Comment by broeder-j** [comment]
Do you plan also to have a `publicdata.yml` at some point?

**Comment by bzg** [comment, marked as answer]
As I understand it, `CITATION.cff` provides what is needed for a software or a dataset to be correctly quoted in scientific papers.

In my view, `publiccode.yml` provides a bit more and is more tailored to the needs of public agencies: for example, it provides the `usedBy` keyword that helps administrations (manually) keep track of what other entities are reusing the software. (See [an example](https://github.com/codegouvfr/codegouvfr-fetch-data/blob/master/publiccode.yml#L12) here.)

Another useful standard, closer to what `publiccode.yml` provides, is [codemeta](https://codemeta.github.io/).

In France, we expect `codemeta` to be used for public research software while `publiccode.yml` will be used for public agencies projects.  That's okay, provided we can align the two schemas together (see https://codemeta.github.io/crosswalk/ for how codemeta matches other schemas - publiccode.yml could be added there.)

In both cases, `CITATION.cff` can complement `codemeta.json` or `publiccode.yml` so that the software is correctly quoted in scientific papers.

**Reply by ruphy** [reply]
I couldn't have said it better @bzg, I completely agree with what you say!

**Comment by broeder-j** [comment]
Thanks a lot for the reply. Now I understand it better.

I was not aware of `codemeta.json` but with the semantics (json-LD) and being close to schema.org. I think this may be it, since this can rather easy be integrated with metadata of other publication types which is out there.
Good that you plan to write a Crosswalk to and from that.

I like the `usedBy`key, through it is free text, non validated right? (Was github not working on such a feature to get this from the whole dependency graph? If this is out a developer could updated `usedBy` in an automatic way. As identifiers one could use the software package names, and for institutions and projects their identifiers. Through without additional metadata this can probably be not validated.

Opinion: `Citation.cff` may become obsolete once the richer metadata standards become better integrated and supported by platforms.

**Comment by bzg** [comment]
> I like the usedBykey, through it is free text, non validated right?  (Was github not working on such a feature to get this from the whole dependency graph? If this is out a developer could updated usedBy in an automatic way. As identifiers one could use the software package names, and for institutions and projects their identifiers. Through without additional metadata this can probably be not validated.

GitHub data can certainly be used to somehow feed the `usedBy` key but I believe there will always be room for manual updates.

> Opinion: Citation.cff may become obsolete once the richer metadata standards become better integrated and supported by platforms.

Scientific software citation is a complex issue and I'm clearly not knowledgeable enough about it to predict anything -- let's see what the scientific community comes up with!
